NashvilleHistory/www/js/factories/MarkerCardsFact.js

```diff
@@ -45,13 +45,45 @@ app.factory("MarkerCardsFact", ($q, $http, KeyGetter)=>{
     return $q((resolve, reject)=>{
       $http.get(`https://maps.googleapis.com/maps/api/distancematrix/json?units=imperial&origins=${lat},${long}&destinations=${markerLat}%2C${markerLong}&key=${GoogleAppToken}`)
       .success((data)=>{
-        resolve(data);
+        // If Google API errors out, calculate distance manually
+        if (data.error_message) {
+          console.log("using manual distance");
+          let distance = calculateDistanceToMarker(lat, long, markerLat, markerLong);
+          resolve(distance);
+        } else {
+          resolve(data);
+        }
       })
       .error((err)=>{
-        reject(err);
+        console.error(error);
+        reject(error);
       })
     });
   }
 
+  function calculateDistanceToMarker(lat1, lon1, lat2, lon2){
+    var R = 6371e3; // metres
+    var φ1 = toRadians(lat1);
+    var φ2 = toRadians(lat2);
+    var Δφ = toRadians((lat2-lat1));
+    var Δλ = toRadians((lon2-lon1));
+
+    var a = Math.sin(Δφ/2) * Math.sin(Δφ/2) +
```

Comment:
Can you put a comment with the formula? I'm assuming all of this works flawlessly. Not sure how to reproduce and test 100% without maxing out my call limit :smile:. LGTM